### PR TITLE
fix(android): Do not show Up navigation button for NavigationWindow's root window.

### PR DESCRIPTION
**Pull request by ypbnv**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26834

**Description:**
Expose the root window for the NavigationWindow's list of windows. Do not show the Up navigation
button for the root window of NavigationWindow's list. Clean a few unused imports. Fixes
TIMOB-26834.
Note: No unit test here, because this is a visual change.

**Test case:**
```js
var win2 = Titanium.UI.createWindow({
    backgroundColor: 'red',
    title: 'Red Window'
});

var win1 = Titanium.UI.createNavigationWindow({
   window: win2
});

var win3 = Titanium.UI.createWindow({
    backgroundColor: 'blue',
    title: 'Blue Window'
});

var button = Titanium.UI.createButton({
    title: 'Open Blue Window'
});
button.addEventListener('click', function(){
    win1.openWindow(win3, {animated:true});
});

win2.add(button);
var button2 = Titanium.UI.createButton({
    title: 'Close Blue Window'
});
button2.addEventListener('click', function(){
    win1.closeWindow(win3, {animated:false}); //win3.close() will also work!!
});

win3.add(button2);
win1.open();
```

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java [hunk 1]
```diff
@@ -93,4 +93,13 @@ public String getApiName()
 	{
 		return "Ti.UI.NavigationWindow";
 	}
+
+	public WindowProxy getRootWindowProxy()
+	{
+		if (!windows.isEmpty()) {
+			return windows.get(0);
+		} else {
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java]: Could remove the `else`?
```Java
if (!windows.isEmpty()) {
  return windows.get(0);
}
return null;
```

File: android/modules/ui/src/java/ti/modules/titanium/ui/WindowProxy.java [hunk 2]
```diff
@@ -286,7 +284,11 @@ public void windowCreated(TiBaseActivity activity, Bundle savedInstanceState)
 			}
 			activity.getSupportActionBar().setHomeButtonEnabled(
 				!getProperties().optBoolean(TiC.PROPERTY_HIDES_BACK_BUTTON, false));
-			activity.getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+			// Get a reference to the root window in the NavigationWindow.
+			WindowProxy rootWindowProxy = ((NavigationWindowProxy) this.getNavigationWindow()).getRootWindowProxy();
+			// If the root window matches this window do not show the Up navigation button.
+			activity.getSupportActionBar().setDisplayHomeAsUpEnabled(rootWindowProxy != null ? rootWindowProxy != this
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/WindowProxy.java]: Could be simplified a little?
```Java
activity.getSupportActionBar().setDisplayHomeAsUpEnabled(!(rootWindowProxy == this));
```

**Comment by ypbnv** [comment]
@garymathews Updated.

**Review by sgtcoolguy — APPROVED** [review]
LGTM, looks like you incorporated @garymathews 's feedback. Marking for qe testing.

**Comment by ssjsamir** [comment]
@ypbnv when I try to build the SDK from this PR I get the following error 
```
[javac] /Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/modules/ui/src/java/ti/modules/titanium/ui/WindowProxy.java:399: error: cannot find symbol
[javac] 					Log.w(TAG, "There is no ActionBar available for this Window.");
    [javac] 					^
    [javac]   symbol:   variable Log
    [javac]   location: class WindowProxy
[javac] Note: Some input files use or override a deprecated API.
[javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
[javac] 1 error
BUILD FAILED
/Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/build/common.xml:658: The following error occurred while executing this line:
/Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/build/common.xml:754: The following error occurred while executing this line:
/Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/build/common.xml:756: The following error occurred while executing this line:
/Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/build/common.xml:733: The following error occurred while executing this line:
/Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/build/common.xml:614: The following error occurred while executing this line:
/Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/build/common.xml:310: The following error occurred while executing this line:
/Users/Samir/Desktop/Appc/Repos/titanium_mobile/android/build/common.xml:183: Compile failed; see the compiler error output for details.

Total time: 19 seconds
Ant build failed with exit code: 1

Something went wrong. Please check console output. Exited with exit code 1.

```

**Review by ssjsamir — APPROVED** [review]
**FR Passed**, NavigationView root window no longer displays back button. 

**Test Environment**
```
Google pixel xl 7.1.1 sim
APPC CLI: 7.0.10
Operating System Name: Mac OS Mojave
Operating System Version: 10.14.2
Node.js Version: 8.9.1
Xcode 10.1
```

**Pull request merged**